### ISSUE TITLE
Script fails to detect duplicate configurations

### ISSUE THREAD

**Issue by menroka** [davidfoerster/aptsources-cleanup#41]
Hello all!

I am not sure what is going wrong, but the script only outputs "No duplicate entries were found." and I am still getting the following output from apt update:

```
sudo apt-get update
Hit:1 http://archive.canonical.com/ubuntu focal InRelease
Get:2 https://dl.winehq.org/wine-builds/ubuntu focal InRelease [8.041 B]       
Hit:4 http://ppa.launchpad.net/diesch/stable/ubuntu focal InRelease            
Err:2 https://dl.winehq.org/wine-builds/ubuntu focal InRelease                 
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 76F1A20FF987672F
Hit:5 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu focal InRelease     
Err:6 https://packages.microsoft.com/ubuntu/20.04/prod focal InRelease         
  Bad header line Bad header data [IP: 13.107.246.67 443]
Hit:7 https://esm.ubuntu.com/infra/ubuntu focal-infra-security InRelease       
Hit:8 http://ppa.launchpad.net/lutris-team/lutris/ubuntu focal InRelease       
Hit:9 https://esm.ubuntu.com/infra/ubuntu focal-infra-updates InRelease        
Hit:10 http://ppa.launchpad.net/micahflee/ppa/ubuntu focal InRelease           
Err:11 https://repo.skype.com/deb stable InRelease                             
  503  Service Unavailable [IP: 23.216.255.8 443]
Hit:3 https://packages.microsoft.com/repos/code stable InRelease    
Hit:12 http://archive.ubuntu.com/ubuntu focal InRelease
Hit:13 http://archive.ubuntu.com/ubuntu focal-updates InRelease
Hit:14 http://archive.ubuntu.com/ubuntu focal-security InRelease
Reading package lists... Done
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: https://dl.winehq.org/wine-builds/ubuntu focal InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 76F1A20FF987672F
W: Failed to fetch https://dl.winehq.org/wine-builds/ubuntu/dists/focal/InRelease  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 76F1A20FF987672F
W: Failed to fetch https://packages.microsoft.com/ubuntu/20.04/prod/dists/focal/InRelease  Bad header line Bad header data [IP: 13.107.246.67 443]
W: Failed to fetch https://repo.skype.com/deb/dists/stable/InRelease  503  Service Unavailable [IP: 23.216.255.8 443]
W: Some index files failed to download. They have been ignored, or old ones used instead.
W: Target Packages (main/binary-amd64/Packages) is configured multiple times in /etc/apt/sources.list:53 and /etc/apt/sources.list.d/winehq-focal.sources:1
W: Target Packages (main/binary-i386/Packages) is configured multiple times in /etc/apt/sources.list:53 and /etc/apt/sources.list.d/winehq-focal.sources:1
W: Target Packages (main/binary-all/Packages) is configured multiple times in /etc/apt/sources.list:53 and /etc/apt/sources.list.d/winehq-focal.sources:1
W: Target Translations (main/i18n/Translation-en_US) is configured multiple times in /etc/apt/sources.list:53 and /etc/apt/sources.list.d/winehq-focal.sources:1
W: Target Translations (main/i18n/Translation-en) is configured multiple times in /etc/apt/sources.list:53 and /etc/apt/sources.list.d/winehq-focal.sources:1
W: Target DEP-11 (main/dep11/Components-amd64.yml) is configured multiple times in /etc/apt/sources.list:53 and /etc/apt/sources.list.d/winehq-focal.sources:1
W: Target DEP-11 (main/dep11/Components-all.yml) is configured multiple times in /etc/apt/sources.list:53 and /etc/apt/sources.list.d/winehq-focal.sources:1
W: Target DEP-11-icons-small (main/dep11/icons-48x48.tar) is configured multiple times in /etc/apt/sources.list:53 and /etc/apt/sources.list.d/winehq-focal.sources:1
W: Target DEP-11-icons (main/dep11/icons-64x64.tar) is configured multiple times in /etc/apt/sources.list:53 and /etc/apt/sources.list.d/winehq-focal.sources:1
W: Target DEP-11-icons-hidpi (main/dep11/icons-64x64@2.tar) is configured multiple times in /etc/apt/sources.list:53 and /etc/apt/sources.list.d/winehq-focal.sources:1
W: Target CNF (main/cnf/Commands-amd64) is configured multiple times in /etc/apt/sources.list:53 and /etc/apt/sources.list.d/winehq-focal.sources:1
W: Target CNF (main/cnf/Commands-all) is configured multiple times in /etc/apt/sources.list:53 and /etc/apt/sources.list.d/winehq-focal.sources:1
```

**Comment by davidfoerster** [comment]
Thanks for the report! Could you please attach `/etc/apt/sources.list` and `/etc/apt/sources.list.d/winehq-focal.sources`? It looks like the sources are configured with restricted architectures and translations and I would like to get my hands on a practical example of those.

Those cases aren't well supported since they're rare enough that I didn't bother to spend much time on a search for realistic test cases.

**Comment by menroka** [comment]
Thank you for taking a look at it!
Here are the requested files:
[sources.list.txt](https://github.com/user-attachments/files/17602074/sources.list.txt)
[winehq-focal.sources.txt](https://github.com/user-attachments/files/17602076/winehq-focal.sources.txt)
added .txt so it would let me upload

I may have screwed it up slightly though, because my entire system kept crashing randomly and I did a whole bunch of modifications since the OP to get it stable again and now I dont get these warnings anymore as a side effect. Sorry about that, Im not sure if the files are still original.

Cheers!

**Comment by davidfoerster** [comment]
> I may have screwed it up slightly though, because my entire system kept crashing randomly and I did a whole bunch of modifications since the OP to get it stable again and now I dont get these warnings anymore as a side effect. Sorry about that, Im not sure if the files are still original.

In that case it would help to see the new output of `apt update`.

**Comment by davidfoerster** [comment]
> [sources.list.txt](https://github.com/user-attachments/files/17602074/sources.list.txt)

This type of Apt sources format didn't exist when I wrote this project and, unfortunately, I don't have a working Debian or Ubuntu installation at hand to test its behaviour.

At least in their current state, I don't see any duplicate sources entries in your two Apt sources files. Maybe the issue is gone after all the changes to your setup.

**Comment by menroka** [comment]
Yep, I think I removed the reference to the winehq server from the sources.list manually, but I dont remember exactly. All duplicates are gone, Current output is:
```
Hit:1 http://archive.canonical.com/ubuntu focal InRelease
Get:2 https://dl.winehq.org/wine-builds/ubuntu focal InRelease [8.041 B]       
Hit:3 http://ppa.launchpad.net/diesch/stable/ubuntu focal InRelease            
Hit:4 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu focal InRelease     
Get:5 https://packages.microsoft.com/ubuntu/20.04/prod focal InRelease [3.632 B]
Hit:6 http://ppa.launchpad.net/micahflee/ppa/ubuntu focal InRelease            
Get:7 https://esm.ubuntu.com/infra/ubuntu focal-infra-security InRelease [7.450 B]
Err:2 https://dl.winehq.org/wine-builds/ubuntu focal InRelease                 
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 76F1A20FF987672F
Err:8 https://repo.skype.com/deb stable InRelease                              
  503  Service Unavailable [IP: 23.55.96.148 443]
Hit:9 http://archive.ubuntu.com/ubuntu focal InRelease                         
Get:10 https://packages.microsoft.com/repos/code stable InRelease [3.590 B]
Get:11 https://esm.ubuntu.com/infra/ubuntu focal-infra-updates InRelease [7.449 B]
Get:12 http://archive.ubuntu.com/ubuntu focal-updates InRelease [128 kB]
Get:13 https://packages.microsoft.com/repos/code stable/main armhf Packages [18,5 kB]
Get:14 https://packages.microsoft.com/repos/code stable/main arm64 Packages [18,4 kB]
Get:15 https://packages.microsoft.com/repos/code stable/main amd64 Packages [18,3 kB]
Get:16 http://archive.ubuntu.com/ubuntu focal-security InRelease [128 kB]
Get:17 http://archive.ubuntu.com/ubuntu focal-updates/main i386 Packages [1.048 kB]
Get:18 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 Packages [3.652 kB]
Get:19 http://archive.ubuntu.com/ubuntu focal-updates/main Translation-en [560 kB]
Get:20 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 DEP-11 Metadata [276 kB]
Get:21 http://archive.ubuntu.com/ubuntu focal-updates/restricted amd64 DEP-11 Metadata [212 B]
Get:22 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 DEP-11 Metadata [446 kB]
Get:23 http://archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 DEP-11 Metadata [940 B]
Get:24 http://archive.ubuntu.com/ubuntu focal-security/main i386 Packages [830 kB]
Get:25 http://archive.ubuntu.com/ubuntu focal-security/main amd64 Packages [3.284 kB]
Get:26 http://archive.ubuntu.com/ubuntu focal-security/main Translation-en [482 kB]
Get:27 http://archive.ubuntu.com/ubuntu focal-security/main amd64 DEP-11 Metadata [65,2 kB]
Get:28 http://archive.ubuntu.com/ubuntu focal-security/restricted amd64 DEP-11 Metadata [212 B]
Get:29 http://archive.ubuntu.com/ubuntu focal-security/universe i386 Packages [682 kB]
Get:30 http://archive.ubuntu.com/ubuntu focal-security/universe amd64 Packages [1.014 kB]
Get:31 http://archive.ubuntu.com/ubuntu focal-security/universe amd64 DEP-11 Metadata [160 kB]
Get:32 http://archive.ubuntu.com/ubuntu focal-security/multiverse amd64 DEP-11 Metadata [940 B]
Fetched 12,8 MB in 19s (684 kB/s)                                              
Reading package lists... Done
Building dependency tree       
Reading state information... Done
3 packages can be upgraded. Run 'apt list --upgradable' to see them.
W: An error occurred during the signature verification. The repository is not updated and the previous index files will be used. GPG error: https://dl.winehq.org/wine-builds/ubuntu focal InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 76F1A20FF987672F
W: Failed to fetch https://repo.skype.com/deb/dists/stable/InRelease  503  Service Unavailable [IP: 23.55.96.148 443]
W: Failed to fetch https://dl.winehq.org/wine-builds/ubuntu/dists/focal/InRelease  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 76F1A20FF987672F
W: Some index files failed to download. They have been ignored, or old ones used instead.
```
